agent/proxycfg/mesh_gateway.go

```diff
@@ -621,7 +621,13 @@ func (s *handlerMeshGateway) handleUpdate(ctx context.Context, u UpdateEvent, sn
 				continue
 			}
 
-			hostnames, ips := peerHostnamesAndIPs(meshLogger, peering.Name, peering.PeerServerAddresses)
+			// Use the manual server addresses, if defined. Otherwise, use the peer addresses.
+			serverAddresses := peering.ManualServerAddresses
+			if len(serverAddresses) == 0 {
+				serverAddresses = peering.PeerServerAddresses
+			}
```

Comment:
Should we put this logic into a helper function? It looks like this is repeated a few times.

Reply:
I can probably put that into a method that's called `GetAddressesToDial()` or something like that.